model/example_domain_test.go

```diff
@@ -0,0 +1,618 @@
+
+	fmt.Println(domain.Empty())
+
+	_, singleton := domain.Value()
+
+	fmt.Println(singleton)
+
+	fmt.Println(domain.Min())
+
+	fmt.Println(domain.Max())
+
+	// Output:
+	// false
+	// false
+	// 0 true
+	// 1 true
+}
+
+func ExampleNewDomain_multipleRange_first() {
+	domain := model.NewDomain(
+		model.NewRange(0, 0),
+		model.NewRange(0, 0),
+	)
+
+	fmt.Println(domain.Empty())
+
+	fmt.Println(domain.Value())
+
+	fmt.Println(domain.Min())
+
+	fmt.Println(domain.Max())
+
+	// Output:
+	// false
+	// 0 true
+	// 0 true
+	// 0 true
+}
+
+func ExampleNewDomain_multipleRange_second() {
+	domain := model.NewDomain(
+		model.NewRange(0, 1),
+		model.NewRange(0, 0),
+	)
+
+	fmt.Println(domain.Empty())
+
+	_, singleton := domain.Value()
+
+	fmt.Println(singleton)
+
+	fmt.Println(domain.Min())
+
+	fmt.Println(domain.Max())
+
+	// Output:
+	// false
+	// false
+	// 0 true
+	// 1 true
+}
+
+func ExampleNewDomain_multipleRange_third() {
+	domain := model.NewDomain(
+		model.NewRange(0, 1),
+		model.NewRange(0, 1),
+	)
+
+	fmt.Println(domain.Empty())
+
+	_, singleton := domain.Value()
+
+	fmt.Println(singleton)
+
+	fmt.Println(domain.Min())
+
+	fmt.Println(domain.Max())
+
+	// Output:
+	// false
+	// false
+	// 0 true
+	// 1 true
+}
+
+func ExampleNewDomain_multipleRange_fourth() {
+	domain := model.NewDomain(
+		model.NewRange(0, 1),
+		model.NewRange(0, 2),
+	)
+
+	fmt.Println(domain.Empty())
+
+	_, singleton := domain.Value()
+
+	fmt.Println(singleton)
+
+	fmt.Println(domain.Min())
+
+	fmt.Println(domain.Max())
+
+	// Output:
+	// false
+	// false
+	// 0 true
+	// 2 true
+}
+
+func ExampleNewDomain_multipleRange_fifth() {
+	domain := model.NewDomain(
+		model.NewRange(0, 2),
+		model.NewRange(0, 1),
+	)
+
+	fmt.Println(domain.Empty())
+
+	_, singleton := domain.Value()
+
+	fmt.Println(singleton)
+
+	fmt.Println(domain.Min())
+
+	fmt.Println(domain.Max())
+
+	// Output:
+	// false
+	// false
+	// 0 true
+	// 2 true
+}
+
+func ExampleNewDomain_multipleRange_sixth() {
+	domain := model.NewDomain(
+		model.NewRange(0, 1),
+		model.NewRange(2, 3),
+	)
+
+	fmt.Println(domain.Min())
+
+	fmt.Println(domain.Max())
+
+	// Output:
+	// 0 true
+	// 3 true
+}
+
+func ExampleNewDomain_multipleRange_seventh() {
+	domain := model.NewDomain(
+		model.NewRange(0, 1),
+		model.NewRange(3, 4),
+	)
+
+	fmt.Println(domain.Min())
+
+	fmt.Println(domain.Max())
+
+	fmt.Println(domain.Contains(2))
+
+	// Output:
+	// 0 true
+	// 4 true
+	// false
+}
+
+func ExampleNewDomain_multipleRange_eighth() {
+	domain := model.NewDomain(
+		model.NewRange(1, 1),
+		model.NewRange(0, 2),
+	)
+
+	fmt.Println(domain.Min())
+
+	fmt.Println(domain.Max())
+
+	// Output:
+	// 0 true
+	// 2 true
+}
+
+func ExampleNewDomain_multipleRange_ninth() {
+	domain := model.NewDomain(
+		model.NewRange(0, 2),
+		model.NewRange(1, 1),
+	)
+
+	fmt.Println(domain.Min())
+
+	fmt.Println(domain.Max())
+
+	// Output:
+	// 0 true
+	// 2 true
+}
+
+func ExampleDomain_Remove_first() {
```

Comment:
I see, so you did attempt to add examples for the interface methods. However, the naming is wrong. If you render the package docs locally, you can realize that these examples are not displayed anywhere. Please see the [linting and testing](https://github.com/nextmv-io/plugins#linting-and-testing) guide:

> When adding new content to the Go Docs, including testable examples, check the rendered package docs locally. Stand on the root of sdk and run:

```bash
$ pkgsite
2022/08/10 16:32:27 Info: Listening on addr http://localhost:8080
```
> Then open up your browser and go to the desired package. You can start from the module itself:

```text
http://localhost:8080/github.com/nextmv-io/sdk
```

To make the example render follow the suggestion.

```suggestion
func ExampleDomain_remove() {
```

I would apply the same pattern as I have commented out before. Instead of having a `first`, `second` etc... variations, have a single test where you can display multiple domains adding some comments so that the user follows... This is so the number of tests also don't become unmanageable. I will only make this comment once as it applies to the rest of the examples that follow in this file.